R/Staphylococcus_aureus_DAP.R

```diff
@@ -0,0 +1,182 @@
+                                   by = c("pathogen" = "Abbreviation"))
+
+# View the resulting merged data
+head(merged_data_pathogens)
+
+
+#filter out rows where pathogen is empty
+cleaned_data <- merged_data_pathogens %>%
+  distinct() %>%
+  filter(!is.na(pathogen))
+
+# Group by Pathogen, Gene, Drug, and Protein.Accession, then summarize Antibiotic information
+summarized_data <- cleaned_data %>%
+  group_by(Pathogen = Pathogen, Gene = gene, Drug = drug, Protein_Accession = Protein.Accession) %>%
+  summarize(Antibiotic_Info = paste(unique(Molecule), collapse = ", ")) %>%
+  arrange(Pathogen, Gene, Drug, Protein_Accession)
+
+# Filter for Staphylococcus aureus and DAP (Bug-Drug of Interest)
+staph_aureus_dap_combinations <- summarized_data %>%
+  filter(Pathogen == "Staphylococcus aureus", Drug == "DAP")
+
+# View the filtered data
+head(staph_aureus_dap_combinations)
+
+
+#Fetch FASTA sequences from Entrez using protein accession
+#Load required packages
+library(rentrez)
+library(XML)
+library(stringr)
+
+
+# Fetch FASTA sequence from Entrez
+fetch_fasta_sequence <- function(protein_accession) {
+  tryCatch({
+    # Fetch the FASTA sequence using Entrez
+    fasta_seq <- rentrez::entrez_fetch(db = "protein",
+                                       id = protein_accession,
+                                       rettype = "fasta",
+                                       retmode = "text")
+
+    if (!is.null(fasta_seq)) {
+      # Ensure the first line starts with ">"
+      if (!grepl("^>", fasta_seq[1])) {
+        fasta_seq[1] <- paste0(">", fasta_seq[1])
+      }
+
+      # Split the sequence into lines
+      lines <- str_split(fasta_seq, "\n")[[1]]
+
+      # Join the lines back together
+      fasta_seq <- paste(lines, collapse = "\n")
+
+      return(fasta_seq)
+    } else {
+      warning(paste("Failed to retrieve FASTA sequence for protein accession:", protein_accession))
+      return(NULL)
+    }
+  }, error = function(e) {
+    warning(paste("Error fetching FASTA sequence for protein accession:", protein_accession, ":", e$message))
+    return(NULL)
+  })
+}
+
+# Loop through staph_aureus_dap_combinations to fetch and save FASTA sequences
+combined_sequences <- character()
+
+for (i in 1:nrow(staph_aureus_dap_combinations)) {
+  # Fetch FASTA sequence for each protein accession
+  protein_accession <- staph_aureus_dap_combinations$Protein_Accession[i]
+  fasta_sequence <- fetch_fasta_sequence(protein_accession)
+
+  if (!is.null(fasta_sequence)) {
+    combined_sequences <- c(combined_sequences, fasta_sequence)
+  }
+}
+
+# Save the combined FASTA sequences
+filename <- "Staphylococcus_aureus_DAP_sequences4.fasta"
+
+writeLines(combined_sequences, filename)
+print(paste("Combined FASTA sequences saved as", filename))
+
+# Print the first few lines of the combined file
+cat(readLines(filename)[1:10], sep = "\n")
+
+
+#Run FASTA Sequences Through MolEvolvR Web App
+
+library("devtools")
+
+devtools::check()
```

Comment:
This is likely what's causing that error you saw before about infinite recursion. Since the package is already being run by `devtools::check()`, running this within the package code is causing it to start a check within the check (and within that check, repeat ad infinitum...) You definitely don't need to call this within code that's part of the package.